CombinatorialGames/Game/Relabelling.lean

```diff
@@ -0,0 +1,330 @@
+    let Hyz : ⟨yl, yr, yL, yR⟩ ≡r ⟨zl, zr, zL, zR⟩ := ⟨L₂, R₂, hL₂, hR₂⟩
+    refine ⟨Equiv.sumCongr L₁ L₂, Equiv.sumCongr R₁ R₂, ?_, ?_⟩ <;> rintro (i | j)
+    · exact (hL₁ i).addCongr Hyz
+    · exact Hwx.addCongr (hL₂ j)
+    · exact (hR₁ i).addCongr Hyz
+    · exact Hwx.addCongr (hR₂ j)
+termination_by _ x _ z => (x, z)
+
+/-- If `w` has the same moves as `x` and `y` has the same moves as `z`,
+then `w - y` has the same moves as `x - z`. -/
+def Relabelling.subCongr {w x y z : PGame} (h₁ : w ≡r x) (h₂ : y ≡r z) : w - y ≡r x - z :=
+  h₁.addCongr h₂.negCongr
+
+/-- `-(x + y)` has exactly the same moves as `-x + -y`. -/
+def negAddRelabelling : ∀ x y : PGame, -(x + y) ≡r -x + -y
+  | ⟨xl, xr, xL, xR⟩, ⟨yl, yr, yL, yR⟩ => by
+    refine ⟨Equiv.refl _, Equiv.refl _, ?_, ?_⟩
+    all_goals
+      exact fun j =>
+        Sum.casesOn j (fun j => negAddRelabelling _ _) fun j =>
+          negAddRelabelling ⟨xl, xr, xL, xR⟩ _
+termination_by x y => (x, y)
+
+/-- `x + y` has exactly the same moves as `y + x`. -/
+def addCommRelabelling : ∀ x y : PGame.{u}, x + y ≡r y + x
+  | mk xl xr xL xR, mk yl yr yL yR => by
+    refine ⟨Equiv.sumComm _ _, Equiv.sumComm _ _, ?_, ?_⟩ <;> rintro (_ | _) <;>
+      · dsimp
+        apply addCommRelabelling
+termination_by x y => (x, y)
+
+/-- `(x + y) + z` has exactly the same moves as `x + (y + z)`. -/
+def addAssocRelabelling : ∀ x y z : PGame.{u}, x + y + z ≡r x + (y + z)
+  | ⟨xl, xr, xL, xR⟩, ⟨yl, yr, yL, yR⟩, ⟨zl, zr, zL, zR⟩ => by
+    refine ⟨Equiv.sumAssoc _ _ _, Equiv.sumAssoc _ _ _, ?_, ?_⟩
+    · rintro (⟨i | i⟩ | i)
+      · apply addAssocRelabelling
+      · apply addAssocRelabelling ⟨xl, xr, xL, xR⟩ (yL i)
+      · apply addAssocRelabelling ⟨xl, xr, xL, xR⟩ ⟨yl, yr, yL, yR⟩ (zL i)
+    · rintro (⟨i | i⟩ | i)
+      · apply addAssocRelabelling
+      · apply addAssocRelabelling ⟨xl, xr, xL, xR⟩ (yR i)
+      · apply addAssocRelabelling ⟨xl, xr, xL, xR⟩ ⟨yl, yr, yL, yR⟩ (zR i)
+termination_by x y z => (x, y, z)
+
+/-- `x * y` and `y * x` have the same moves. -/
+def mulCommRelabelling (x y : PGame.{u}) : x * y ≡r y * x :=
+  match x, y with
+  | ⟨xl, xr, xL, xR⟩, ⟨yl, yr, yL, yR⟩ => by
+    refine ⟨Equiv.sumCongr (Equiv.prodComm _ _) (Equiv.prodComm _ _),
+      (Equiv.sumComm _ _).trans (Equiv.sumCongr (Equiv.prodComm _ _) (Equiv.prodComm _ _)), ?_, ?_⟩
+      <;>
+    rintro (⟨i, j⟩ | ⟨i, j⟩) <;>
+    { dsimp
+      exact ((addCommRelabelling _ _).trans <|
+        (mulCommRelabelling _ _).addCongr (mulCommRelabelling _ _)).subCongr
+        (mulCommRelabelling _ _) }
+  termination_by (x, y)
+
+/-- `x * 0` has exactly the same moves as `0`. -/
+def mulZeroRelabelling (x : PGame) : x * 0 ≡r 0 :=
+  Relabelling.isEmpty _
+
+/-- `0 * x` has exactly the same moves as `0`. -/
+def zeroMulRelabelling (x : PGame) : 0 * x ≡r 0 :=
+  Relabelling.isEmpty _
+
+/-- `-x * y` and `-(x * y)` have the same moves. -/
+def negMulRelabelling (x y : PGame.{u}) : -x * y ≡r -(x * y) :=
```

Comment:
I slightly golfed this def in the PR adressing porting notes, you could copy it over (even though it really doesn't matter much).